project/apps/web/app/question/[id]/page.tsx

```diff
@@ -0,0 +1,179 @@
+"use client";
+
+import { Suspense } from "react";
+import {
+  useSuspenseQuery,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { notFound, useRouter } from "next/navigation";
+import {
+  fetchQuestionById,
+  updateQuestion,
+  deleteQuestion,
+} from "@/lib/api/question";
+import { QuestionDto, UpdateQuestionDto } from "@repo/dtos/questions";
+import Link from "next/link";
+import { Button } from "@/components/ui/button";
+import { ArrowLeft, Pencil, Trash2 } from "lucide-react";
+import DifficultyBadge from "@/components/DifficultyBadge";
+import { Badge } from "@/components/ui/badge";
+import { QUERY_KEYS } from "@/constants/queryKeys";
+import QuestionSkeleton from "./components/QuestionSkeleton";
+import EditModal from "./components/EditModal";
+import DeleteModal from "./components/DeleteModal";
+import { useState } from "react";
+import { toast } from "@/hooks/use-toast";
+
+interface QuestionPageProps {
+  params: {
+    id: string;
+  };
+}
+
+const QuestionPageContent = ({ id }: { id: string }) => {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+  const [isEditModalOpen, setEditModalOpen] = useState(false);
+  const [isDeleteModalOpen, setDeleteModalOpen] = useState(false);
+  const [confirmLoading, setConfirmLoading] = useState(false);
+
+  const { data: question } = useSuspenseQuery<QuestionDto>({
+    queryKey: [QUERY_KEYS.Question, id],
+    queryFn: () => fetchQuestionById(id),
+  });
+
+  const updateMutation = useMutation({
+    mutationFn: (updatedQuestion: UpdateQuestionDto) =>
+      updateQuestion(updatedQuestion),
+    onMutate: () => setConfirmLoading(true),
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
+        queryKey: [QUERY_KEYS.Question, id],
+      });
+      setEditModalOpen(false);
+      toast({ variant: "success", title: "Success", description: "Question updated successfully" });
+    },
+    onSettled: () => setConfirmLoading(false),
+    onError: (error) => {
+      console.error("Error updating question:", error);
+      toast({ variant: "destructive", title: "Error", description: "Error updating question" + error });
+    },
```

Comment:
```suggestion
    onError: (error) => {
      const errorMsg = `Error updating question: ${error.message}`;
      console.error(errorMsg);
      toast({
        variant: "destructive",
        title: "Error",
        description: errorMsg,
      });
    },
  });
```

Before:
<img width="450" alt="Screenshot 2024-09-29 at 5 06 53 PM" src="https://github.com/user-attachments/assets/e8ee1ce7-dd53-4b4f-a3bb-c02000e42658">
After:
<img width="435" alt="Screenshot 2024-09-29 at 5 05 44 PM" src="https://github.com/user-attachments/assets/d7e4b11d-1a32-4037-8112-8f6d9fdbccc5">

Reply:
`"Error updating question" + error`: its a string concatenation here, so no whitespace is inserted